src/buf.c

```diff
@@ -266,4 +285,10 @@ struct Buf *buf_append (struct Buf *buf, const void *ptr, int n_elem)
 	return buf;
 }
 
+struct Buf *buf_append_alloc (struct Buf *buf, const void *ptr, int n_elem)
+{
+	buf_append(&alloc_buf, ptr, n_elem);
```

Comment:
So the problem you are trying to address is a `Buf` of allocated strings? (Similar to `std::vector<const char *>` if I use C++ terminology.)

I don't think your solution is the best way, because, what if I change a single `char` pointer (element) to point to another string without freeing the previous string? It's also memory leak that way.

I think the better way is to force all  `Buf` of strings instances to be responsible to their own string allocation. Don't use a global "things to free()" buffer. Instead implement proper finalize() or destructor for the `Buf` of strings case.